### PR TITLE
Checksums for configs and Nodeselector, affinity, and tolerations

### PR DESCRIPTION
**Pull request by lefes**
Hello!
I would like to help with developing this helm chart and added some useful features like:

- Checksums in statefulset to recreate container when some configurations are changed
- Additional configurations via nodeselector, tolerations, affinity

### CODE REVIEW

**Comment by lefes** [comment]
Silly me, i was in a hurry and put labels on statefulset level. Now its on it right place.

**Comment by guerzon** [comment]
Thanks for the MR @lefes!

File: values.yaml [hunk 1]
```diff
@@ -241,3 +241,21 @@ initContainers: []
 ## @param sidecars extra containers running alongside the vaultwarden instance
 ##
 sidecars: []
+
+## @section Extra Configuration
+##
+
+## @param nodeSelector Node labels for pod assignment
+## Ref: https://kubernetes.io/docs/user-guide/node-selection/
```
Comment (guerzon) [inline values.yaml:249]: Please review the URL, it appears to be broken at the moment. Thanks.

File: templates/statefulset.yaml [hunk 2]
```diff
@@ -16,8 +16,22 @@ spec:
     metadata:
       labels:
         app: vaultwarden
+        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha1sum }}
```
Comment (guerzon) [inline templates/statefulset.yaml:19]: I like the idea of adding `checksum/config` and `checksum/secret` to trigger a restart whenever the configs are updated. However, these are `metadata` annotations and not labels:

```yaml
...
    metadata:
      labels:
        <>
      annotations:
        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha1sum }}
        checksum/secret: {{ include (print $.Template.BasePath "/secrets.yaml") . | sha1sum }}
...
```

Thanks!

**Comment by guerzon** [comment]
Thanks again @lefes for the MR. Aside from the open conversations, please also:

1. Rebase/pull from master.
2. Bump the chart version (ex. from `0.7.0` to `0.8.0`).

**Review by guerzon — CHANGES_REQUESTED** [review]
Thanks again @lefes for the MR. Aside from the open conversations, please also:

1. Rebase/pull from master.
2. Bump the chart version (ex. from `0.7.0` to `0.8.0`).

File: values.yaml (hunk 1 above)
Comment (lefes) [reply values.yaml]: @guerzon thank you for your attention to details. I've fix it

File: templates/statefulset.yaml (hunk 2 above)
Comment (lefes) [reply templates/statefulset.yaml:21]: @guerzon Thank you for pointing that out. You are right. It should be in annotations. Fixed it in last commits :)

**Comment by lefes** [comment]
All changes are complete. Hope i've done all right

**Comment by guerzon** [comment]
Thanks @lefes

**Pull request merged**